apps/core-app/src/contexts/DaoContext.tsx

```diff
@@ -0,0 +1,327 @@
+import { Keychain } from '@daohaus/common-utilities';
+import {
+  DaoWithTokenDataQuery,
+  ITransformedProposalListQuery,
+  ListMembersQuery,
+  Member_Filter,
+  Member_OrderBy,
+  Ordering,
+  Paging,
+  Proposal_Filter,
+  Proposal_OrderBy,
+} from '@daohaus/dao-data';
+import {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+  SetStateAction,
+  Dispatch,
+} from 'react';
+import { useParams } from 'react-router-dom';
+import {
+  loadDao,
+  loadMembersList,
+  loadProposalsList,
+} from '../utils/contextHelpers';
+
+// TODO: maybe better to send next page in a different way - as it's own state element
+
+export const defaultDaoData = {
+  dao: null,
+  isDaoLoading: false,
+  refreshDao: async () => undefined,
```

Comment:
We'd likely want to use void for these so that we're saying they're not returning something as opposed to saying they'll always return undefined -- I'm wondering if undefined may have some consequences down the road. looking into this a bit more because I'm curious

Reply:
The types return void. These are the default values passed into the context replaced by the setState function later.

Maybe I could do this:
```
  refreshMembers: async () => {
    return;
  },
  ```